nextjs-app/src/pages/integrations/form.tsx

```diff
@@ -39,19 +48,21 @@ export default function Form() {
     
         setFavouriteNumberValidationMessage(validationMessage);
     };
-    
-    
 
-    const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-        setEmail(event.target.value);
+    const handleEmailInput = (event: InputEvent) => {
```

Comment:
I'm not sure if this would be clearer or not (or even work like I think it may), but seeing the repetition made me think of doing something like this:

```js
const handleInput = (handlerFn) => {
    return handlerFn((event.target as HTMLInputElement).value);
}

const handleEmailInput = handleInput(setEmail);
const handleUrlInput = handleInput(setUrl);
const handleTelInput = handleInput(setTel);
const handlePasswordInput = handleInput(setPassword);
```

Reply:
maybe in a future PR :)